src/java.base/share/classes/java/lang/AbstractStringBuilder.java

```diff
@@ -1821,4 +1821,97 @@ private final void appendChars(CharSequence s, int off, int end) {
         }
         count += end - off;
     }
+
+    /**
+     * Appends {@code count} copies of the character {@code c} to this sequence.
+     * <p>
+     * The length of this sequence increases by {@code count}.
+     *
+     * @param c      character to append
+     * @param count  number of times to copy
+     *
+     * @return  a reference to this object.
+     *
+     * @since 21
+     * @throws IllegalArgumentException  if {@code count} is less than zero
+     * @throws IndexOutOfBoundsException  if the result overflows the buffer
+     */
+    public AbstractStringBuilder repeat(char c, int count) {
+        if (count < 0) {
+            throw new IllegalArgumentException("count is less than zero: " + count);
+        } else if (count == 0) {
+            return this;
+        }
+        ensureCapacityInternal(this.count + count);
+        int index = this.count;
+        int limit = index + count;
+        boolean isLatin1 = isLatin1();
+        if (isLatin1 && StringLatin1.canEncode(c)) {
+            while (index < limit) {
+                value[index++] = (byte)c;
+            }
+        } else {
+            if (isLatin1) {
+                inflate();
+            }
+            while (index < limit) {
+                StringUTF16.putCharSB(value, index++, c);
+            }
+        }
+        this.count = limit;
+        return this;
+    }
+
+    /**
+     * Appends {@code count} copies of the specified {@code CharSequence} {@code cs}
+     * to this sequence.
+     * <p>
+     * The length of this sequence increases by {@code count} times the
+     * {@code CharSequence} length.
+     *
+     * @param cs     a {@code CharSequence}
+     * @param count  number of times to copy
+     *
+     * @return  a reference to this object.
+     *
+     * @since 21
+     * @throws IllegalArgumentException  if {@code count} is less than zero
+     * @throws IndexOutOfBoundsException  if the result overflows the buffer
```

Comment:
Can `IndexOutOfBoundsException` be really thrown? We don't have any index here. From the implementation I see that only `OutOfMemoryError` and `IllegalArgumentException` can be thrown.

Also, I see a small descrepancy between messages in this method and `String.repeat`. The latter has:

```java
     * @throws  IllegalArgumentException if the {@code count} is
     *          negative.
     *
     * @since 11
     */
    public String repeat(int count) {
        if (count < 0) {
            throw new IllegalArgumentException("count is negative: " + count);
        }
```

Probably it's better to reuse the same wording ("is negative" instead of "is less than zero") for consistency?

Reply:
You are correct about `IndexOutOfBoundsException`. Earlier versions used methods that threw `IndexOutOfBoundsException`. Also, changing to "is negative"